R/pjn_metadata.R

```diff
@@ -46,3 +49,17 @@ get_pjn_region <- function(pjn) {
     return(region)
   }
 }
+
+#' Get subnational region code from parsed PJN
+#'
+#' @keywords internal
+get_pjn_region_code <- function(pjn) {
+  idx <- which(pjn[, "Tag"] == "<Projection Parameters - Subnational Region Name2>") + 3
+  region_code <- pjn[idx, "Data"]
+
+  if (region_code == "") {
+    return(NULL)
```

Comment:
This touches on one of things we talked about during the PDRP about writing idiomatic R code. Using a `return()`, isn't the most idiomatic R way of doing things. I probably didn't know that when I wrote the block above this. 

Do you want to remove the return statements here (and in the function above) to just return the last line. 

```
get_pjn_region_code <- function(pjn) {
  idx <- which(pjn[, "Tag"] == "<Projection Parameters - Subnational Region Name2>") + 3
  region_code <- pjn[idx, "Data"]

  if (region_code == "") {
    region_code <- NULL
  } else {
    region_code <- as.integer(region_code)
  }

  region_code
}
```

and

```
get_pjn_region <- function(pjn) {
  idx <- which(pjn[, "Tag"] == "<Projection Parameters - Subnational Region Name2>") + 2
  region <- pjn[idx, "Data"]
  if (region == "") {
    region <- NULL
  } 
  region
}
```

Reply:
Thanks. I have updated as this and also to return NA instead of NULL, which I think is better practice in R (0023fb98).

`get_pjn_region_code()` updated to return 0 if there is no region, which looks to be the default value for national value in Spectrum.

Reply:
That's good, yeah just did a bit of reading on the NA vs NULL thing. Looks like NA is the way to go here. I see you put a comment in the details section of `get_pjn_region_code` about why we use `0` as a default. That's good, I duplicated that note just above where we set it too so that the explanation is close to the code.